zebra-consensus/src/block/subsidies.rs

```diff
@@ -0,0 +1,241 @@
+//! Calculation of Block Subsidy, Funding Streams, and Founders’ Reward
+
+use super::*;
+
+use zebra_chain::{
+    block::Height,
+    parameters::{Network, NetworkUpgrade::*},
+};
+
+use crate::parameters::{fs, Params};
+
+fn slow_start_shift() -> Height {
+    Height(Params::SLOW_START_INTERVAL / 2)
+}
+
+fn slow_start_rate() -> u32 {
+    Params::MAX_BLOCK_SUBSIDY / Params::SLOW_START_INTERVAL
+}
+
+fn is_bolossom_activated(height: block::Height, network: Network) -> Option<Height> {
+    let blossom_height = Blossom.activation_height(network)?;
+    let condition = height >= blossom_height;
+    match condition {
+        true => Some(blossom_height),
+        false => None,
+    }
+}
+
+fn halving(height: Height, network: Network) -> u32 {
+    let blossom_height = is_bolossom_activated(height, network);
+    let condition = height < slow_start_shift();
+    match condition {
+        true => 0,
+        false => match blossom_height {
+            Some(blossom_height) => {
+                let scaled_halvings = ((blossom_height.0 - slow_start_shift().0)
+                    * Params::BLOSSOM_POW_TARGET_SPACING_RATIO)
+                    + (height.0 - blossom_height.0);
+                scaled_halvings / Params::POST_BLOSSOM_HALVING_INTERVAL
+            }
+            _ => {
+                (((height.0 - slow_start_shift().0) / Params::PRE_BLOSSOM_HALVING_INTERVAL) as f32)
```

Comment:
Should we impl `Sub` for `Height`? cc @ZcashFoundation/zebra-team

Reply:
Yes, I'll make an issue for this.

Reply:
Opened https://github.com/ZcashFoundation/zebra/issues/1054

Reply:
Hey, sorry, I wrote up https://github.com/ZcashFoundation/zebra/issues/1055 yesterday but I forgot to file it and left it open in a tab.

Reply:
Used this at https://github.com/ZcashFoundation/zebra/pull/1051/commits/2ab938fb6f520fb3ee468eaee973e7344b283aab